### PR TITLE
refactor: simplify LeJEPA loss and module surface

### PR DESCRIPTION
**Pull request by gabrielfruet**
related to https://github.com/lightly-ai/lightly/issues/1888

## Description
- [x] My change is breaking

This PR starts addressing the LeJEPA follow-up work in #1888.

Changes:
- aligned LeJEPA defaults and behavior more closely with the paper
- updated the loss to support separate local views and global views
- removed the unnecessary `LeJEPAEncoder` wrapper to reduce coupling and keep the module surface focused on reusable building blocks

## Tests
- [x] My change is covered by existing tests
- [x] I have added/adapted the tests accordingly.

## Documentation
- [x] I have added docstrings to all changed/added public functions/methods.

## Improvements put into another issue:
- #1888

## Issues covering the breaking change:
- https://github.com/lightly-ai/lightly/issues/1888

### CODE REVIEW

File: lightly/loss/lejepa_loss.py [hunk 1]
```diff
@@ -8,27 +8,57 @@
 from lightly.utils import dist as lightly_dist
 
 
-def lejepa_invariance_loss(proj: Tensor) -> Tensor:
+def lejepa_invariance_loss(*, local_proj: Tensor, global_proj: Tensor) -> Tensor:
```
Comment (liopeer) [inline lightly/loss/lejepa_loss.py:11]: Thanks for the kwarg-only implementation! ❤️ This is always so much easier to maintain in the future.

File: lightly/loss/lejepa_loss.py [hunk 2]
```diff
@@ -38,7 +68,7 @@ def __init__(
         self,
         knots: int = 17,
```
Comment (liopeer) [inline lightly/loss/lejepa_loss.py:70]: Btw, here it's also always helpful to only allow kwargs.

File: lightly/loss/lejepa_loss.py [hunk 3]
```diff
@@ -236,15 +268,19 @@ def __init__(
...
-            proj: Projected embeddings of shape ``(V, N, D)``.
+            local_proj: Local-view projected embeddings of shape ``(Vl, N, D)``.
+            global_proj: Global-view projected embeddings of shape ``(Vg, N, D)``.
         """
-        sigreg_loss = self.sigreg(proj)
-        inv_loss = lejepa_invariance_loss(proj)
-        loss: torch.Tensor = (
+        _validate_projection_shapes(local_proj=local_proj, global_proj=global_proj)
+        sigreg_loss = self.sigreg(local_proj)
+        inv_loss = lejepa_invariance_loss(
```
Comment (liopeer) [inline lightly/loss/lejepa_loss.py:282]: Now you're validating the projection shapes twice which is a bit overkill.

File: tests/loss/test_lejepa_loss.py [hunk 4]
```diff
@@ -1,3 +1,5 @@
+from typing import Tuple
```
Comment (liopeer) [inline tests/loss/test_lejepa_loss.py]: Prefer using `tuple` and `from __future__ import annotations`.

**Review by liopeer — APPROVED** [review]
LGTM with minor comments. None of them absolutely have to be addressed. Feel free to address the ones you find valid and resolve yourself!

File: lightly/loss/lejepa_loss.py (hunk 1 above)
Comment (gabrielfruet) [reply lightly/loss/lejepa_loss.py:11]: When I was writing the example I managed to swap both 🤣. So for no one to fall for the same mistake, I did the kwargs only
